### DISCUSSION TITLE
FTP deployment and .scss

### DISCUSSION THREAD

**Discussion by Verickt** [statamic/cms#8570] [Troubleshooting] [answered]
Hello everybody,

I am new to statamic and am trying to deploy my site to a shared hosting via FTP.

I can npm run dev and npm run build locally without an issue. But when I upload it to the FTP: http://aky-p.rcbr.ch/ it tries to fetch the site.scss file. 

Should a production site not only serve the compiled files in the /public/build directory?

I feel like I am missing out on some very essential information. I can't seem to find any ressources on it either. I am using the peak starter kit and added sass to it.

**Comment by duncanmcclean** [comment]
Let's see your layout file and your Vite config.

**Reply by Verickt** [reply]
```

import laravel from 'laravel-vite-plugin'
import { defineConfig, loadEnv } from 'vite'
import { networkInterfaces } from 'os'
import sass from 'vite-plugin-sass';
import VitePluginBrowserSync from 'vite-plugin-browser-sync'

/** @type {import('vite').UserConfig} */
export default defineConfig(({ command, mode }) => {
    const env = loadEnv(mode, process.cwd(), '')
    const ip = Object.values(networkInterfaces()).flat().find((i) => i.family == 'IPv4' && !i.internal)?.address
    const isSecuredSite = env.APP_URL.startsWith('https://')
    const url = env.APP_URL
    const host = url.replace(/^https?:\/\//, '')
    const useBrowserSync = env.VITE_BROWSERSYNC && ip

    if (useBrowserSync && isSecuredSite) {
        console.log('\x1b[31m%s\x1b[0m', 'Sorry, we didn\'t manage to get BrowserSync working with secured sites.');
        console.log('\x1b[31m%s\x1b[0m', 'If you can provide a solution, please enlighten us (https://github.com/studio1902/statamic-peak/pulls).');
        process.exit(1)
    }

    return {
        base: './',
        plugins: [
            laravel({
                refresh: true,
                valetTls: isSecuredSite ? host : false,
                input: [
                    'resources/css/site.scss',
                    'resources/js/site.js',
                ]
            }),
            (() => {
                return useBrowserSync ? VitePluginBrowserSync({
                    bs: {
                        proxy: url,
                        notify: false,
                        open: 'external',
                    },
                }) : null;
            })(),

        ],
        server: {
            https:true,
            host: useBrowserSync ? ip : host,
            open: useBrowserSync ? false : url,
        }
    }
});
```


```

{{# @name Layout @desc The default layout file. #}}

<!-- /layout.antlers.html -->
<!doctype html>
<html lang="{{ site:short_locale }}" class="antialiased">
    <head>
        <meta charset="utf-8" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta
            name="viewport"
            content="width=device-width, initial-scale=1.0, viewport-fit=cover"
        />
        {{ vite src="resources/css/site.scss|resources/js/site.js" }} {{
        partial:statamic-peak-seo::snippets/seo }} {{
        partial:statamic-peak-browser-appearance::snippets/browser_appearance }}
        {{ partial:statamic-peak-tools::snippets/live_preview }}
    </head>
    <body
        class="flex flex-col min-h-screen selection:bg-primary selection:text-white aky-body"
    >
        {{ partial:statamic-peak-tools::snippets/noscript }} {{
        partial:statamic-peak-tools::navigation/skip_to_content }} {{
        partial:statamic-peak-tools::components/toolbar }} {{ yield:seo_body }}
        {{ partial:layout/header }} {{ template_content }} {{
        partial:layout/footer }}
    </body>
</html>
<!-- End: /layout.antlers.html -->

```

**Comment by jasonvarga** [comment]
You should run `npm run build` before deploying to production which puts all the files into `public/build/assets`. I think maybe you're uploading the dev version?

**Reply by Verickt** [reply]
I am doing so. I am then uploading the whole project to the FTP. Should I just be uploading the public directory?

**Comment by Verickt** [comment]
So I can access /cp but clearing all caches in the Utilities did not work either. The issue seems to be that it tries to fetch:
- https://aky-p.rcbr.ch/resources/js/site.js
- https://aky-p.rcbr.ch/resources/css/site.scss

when it should be fetching:
- site-53d3eb58.js
- site-0f59cd8.css
from the /public/build directory.

The .env on the server says production and debug=false but I still see the debug bar on production.

**Comment by Verickt** [comment, marked as answer]
I solved the issue. In my layout.antlers.html I added:

`    <link rel="stylesheet" href="{{ vite:asset src="resources/css/site.scss" }}">
    <script src="{{ vite:asset src="resources/js/site.js" }}"></script>
`

and I created the tag like this:
`<?php

namespace App\Tags;
use \Statamic\Tags\Tags;
class ViteAssets extends Tags
{
    protected static $handle = 'vite';

    public function asset()
    {
        $asset = $this->params->get('src');

        $manifestPath = public_path('build/manifest.json');
        $manifest = json_decode(file_get_contents($manifestPath), true);

        if (isset($manifest[$asset]['file'])) {
            return '/build/' . $manifest[$asset]['file'];
        }

        return null;
    }

}
`

**Reply by jasonvarga** [reply]
You know we have a vite tag built-in, right?
https://statamic.dev/tags/vite

Was it not doing something you needed?
